pkg/ksync/cli.go

```diff
@@ -11,51 +11,159 @@ import (
 	"github.com/jenkins-x/jx/pkg/util"
 )
 
-// InstallKSync install ksync
-func InstallKSync() (string, error) {
+var commit string
+var binary = "ksync"
+
+// CLI implements the command ksync commands contained in KSyncer.
+type CLI struct {
+	Runner util.Commander
+}
+
+// NewCLI creates a new KsyncCLI instance configured to use the provided ksync CLI in the given current working directory
+func NewCLI() (*CLI, error) {
+	path, err := util.JXBinLocation()
+	if err != nil {
+		return nil, err
+	}
+	runner := &util.Command{
+		Name: fmt.Sprintf("%s/ksync", path),
+	}
+	cli := &CLI{
+		Runner: runner,
+	}
+	return cli, nil
+}
+
+// GetJXBinLocation returns the jx bin location
+func getJXBinLocation() (string, error) {
 	binDir, err := util.JXBinLocation()
 	if err != nil {
 		return "", err
 	}
-	binary := "ksync"
-	fileName, flag, err := packages.ShouldInstallBinary(binary)
-	if err != nil || !flag {
-		// Exec `ksync` to find the version
-		ksyncCmd := util.Command{
-			Name: fileName,
-			Args: []string{
-				"version",
-			},
-		}
-		// Explicitly ignore any errors from ksync version, as we just need the output!
-		res, _ := ksyncCmd.RunWithoutRetry()
-		lines := strings.Split(res, "\n")
-		for _, line := range lines {
-			line = strings.TrimSpace(line)
-			if strings.HasPrefix(line, "Git Tag:") {
-				return strings.TrimSpace(strings.TrimPrefix(line, "Git Tag:")), nil
-			}
-		}
+	return binDir, nil
+}
 
-		return "", fmt.Errorf("unable to find version of ksync")
-	}
-	latestVersion, err := util.GetLatestVersionFromGitHub("vapor-ware", "ksync")
+// isInstalled checks if the binary is installed
+func isInstalled(binary string) (bool, error) {
+	_, flag, err := packages.ShouldInstallBinary(binary)
+	return flag, err
+}
+
+// getTag returns the latest full tag from github given the owner and repo, and installs
+func getTag(owner, repo string) (string, string, error) {
+	latestTag, err := util.GetLatestFullTagFromGithub(owner, repo)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
-	clientURL := fmt.Sprintf("https://github.com/vapor-ware/ksync/releases/download/%s/ksync_%s_%s", latestVersion, runtime.GOOS, runtime.GOARCH)
+	latestVersion := *latestTag.Name
+	// This will return the shortsha, the first 7 characters associated with the git commit
+	latestSha := (*latestTag.Commit.SHA)[:7]
+	return latestSha, latestVersion, nil
+}
+
+// install downloads and installs the ksync package
+func install(latestVersion, binDir, fileName string) error {
```

Comment:
I had a test for this, but then thought it was redundant because TestKsyncCMD was testing it indirectly. I can always add it back. Testing the install for all the OS would be nice :).